Sources/GRPC/LengthPrefixedMessageWriter.swift

```diff
@@ -49,9 +49,8 @@ internal struct LengthPrefixedMessageWriter {
   /// - Returns: A `ByteBuffer` containing a gRPC length-prefixed message.
   /// - Precondition: `compression.supported` is `true`.
   /// - Note: See `LengthPrefixedMessageReader` for more details on the format.
-  func write(_ payload: GRPCPayload, into buffer: inout ByteBuffer, disableCompression: Bool = false) throws {
-    buffer.reserveCapacity(MemoryLayout.size(ofValue: payload) + LengthPrefixedMessageWriter.metadataLength)
-    
+  func write(_ payload: GRPCPayload, into buffer: inout ByteBuffer, disableCompression: Bool = false) throws {r
+    buffer.reserveCapacity(buffer.writerIndex + LengthPrefixedMessageWriter.metadataLength)
```

Comment:
moved the code to use the `buffer.writerIndex` instead of the `MemoryLayout.size(of: payload)` since I think this way it will be more robust knowing that it will always have those 5 extra bytes on the buffer.

Reply:
Great catch, thanks!